spring-pulsar/src/test/java/org/springframework/pulsar/core/PulsarTemplateTests.java

```diff
@@ -118,6 +120,27 @@ void interceptorInvocationTest(String topic, List<ProducerInterceptor> intercept
 		}
 	}
 
+	@Test
+	void sendMessageWithSpecificSchemaTest() throws Exception {
+		String topic = "smt-specific-schema-topic";
+		try (PulsarClient client = PulsarClient.builder().serviceUrl(getPulsarBrokerUrl()).build()) {
+			try (Consumer<Foo> consumer = client.newConsumer(Schema.JSON(Foo.class)).topic(topic)
+					.subscriptionName("test-specific-schema-subscription").subscribe()) {
+				PulsarProducerFactory<Foo> producerFactory = new DefaultPulsarProducerFactory<>(client,
+						Collections.singletonMap("topicName", topic));
+				PulsarTemplate<Foo> pulsarTemplate = new PulsarTemplate<>(producerFactory);
+				pulsarTemplate.setSchema(Schema.JSON(Foo.class));
+				Foo foo = new Foo("Foo-" + UUID.randomUUID(), "Bar-" + UUID.randomUUID());
+				pulsarTemplate.send(foo);
+
+				CompletableFuture<Message<Foo>> receiveMsgFuture = consumer.receiveAsync();
+				Message<Foo> msg = receiveMsgFuture.get(3, TimeUnit.SECONDS);
```

Comment:
Have you seen AssertJ support for `CompletableFuture`? It is pretty handy. Here are some [examples](https://github.com/assertj/assertj-examples/blob/main/assertions-examples/src/test/java/org/assertj/examples/CompletableFutureAssertionsExamples.java).